src/command.rs

```diff
@@ -228,10 +233,15 @@ impl EccCommand {
                 let mut param1 = SignParam(0);
                 param1.set_source(source.into());
                 param1.set_external(true);
-                put_cmd!(bytes, ATCA_SIGN, u8::from(param1), (*key_slot as u16) << 8);
+                put_cmd!(
+                    bytes,
+                    ATCA_SIGN,
+                    u8::from(param1),
+                    (*key_slot as u16).to_be()
+                );
             }
             Self::Ecdh { x, y, key_slot } => {
-                put_cmd!(bytes, ATCA_ECDH, 0, *key_slot as u16);
+                put_cmd!(bytes, ATCA_ECDH, 0, (*key_slot as u16).to_be());
```

Comment:
This is the weird one though.. Why is this not getting written as a bit endian by put_u16 to begin with? There's definitely a bug but I'm not sure this is the correct fix for it

Reply:
From the ecc608 documentation: 

```16 bit (2 byte) integers, typically Param2, SlotConfig or KeyConfig, appear on the bus least- significant byte first.```

Reply:
so perhaps the fix here is to no do any to_be or byte shifting and just change `put_cmd` to use `put_u16_le`?

Reply:
Well, If I change put_u16 to put_u16_le in put_cmd, all the commands passing u16::from(address) as param2 are not working anymore so fix should include all bitfield correction. 
Really weird as you say, I need also to dive into it a bit more if you are not sure my fix is generic.

Reply:
Yeah I'm concerned put_cmd isn't doing the right thing with param2 there.. if the bug is real wouldn't key and slot config u16s also be wrong? (as per the microchip docs)

Reply:
Ho I remember what I saw on the bus, sending the command ECDH with slot 3:
- gateway-mfr-rs:
  03474300000380360537...
- my application based on CryptoAuthlib:
  03474300030080360537...

The correct solution is indeed little endian and put_u16_le but then Address bitfield is not correct anymore (works in my case if I use u16::from(address).swap_bytes). It's like bitfield is already le, is it possible ?

Reply:
ok so what about leaving sign and genkey commands as they were (since << 8 already does the right thing) and making the smallest possible fix with the same bit shift for ecdh?

Reply:
After some reading in the datasheet (Address encoding data zone for exemple), it appears indeed the address bitfield is already formatted in little endian.

Regarding what you propose, it's up to you. I can fix it like that for now but I will probably try to clean it on my spare time.

Reply:
hmm though I guess  (x << 8) as u16 is assuming the current machine is in little endian format. Which is certainly the current case for all hardware but not ideal to make that assumption

Reply:
My though is bitfields should be represented exactly as in the datasheet no matter the endianess while it should be converted to bus endianess (little in this case) so my plan is to use put16_le for param2 and rewrite all the bitfields, I keep you posted next week

Reply:
That seems like a bit of work.. In the meantime I'll push up #14 to get helium-crypto to do the right thing for ecc608 ecdh

Reply:
Ok, I changed for a quick and safe fix that you can push. I will work on another branch and pull request for the cleaner fix so that's not blocking.